src/wfdirrd.c

```diff
@@ -1227,6 +1227,124 @@ DWORD DecodeReparsePoint(LPCWSTR szMyFile, LPCWSTR szChild, LPWSTR szDest, DWORD
 	return reparseTag;
 }
 
+#define PATH_PARSE_SWITCHOFF L"\\\\?\\" 
+#define PATH_PARSE_SWITCHOFF_SIZE (sizeof(PATH_PARSE_SWITCHOFF) - 1) / sizeof(wchar_t)
+#define REPARSE_MOUNTPOINT_HEADER_SIZE   8
+
+BOOL IsVeryLongPath(LPCWSTR pszPathName)
+{
+   return (wcslen(pszPathName) >= COUNTOF(PATH_PARSE_SWITCHOFF) - 1) && !wcsncmp(pszPathName, PATH_PARSE_SWITCHOFF, COUNTOF(PATH_PARSE_SWITCHOFF) - 1);
+}
+
+/* WFJunction
+ *
+ * Creates a NTFS Junction
+ * Returns either ERROR_SUCCESS or GetLastError()
+ */
+DWORD WFJunction(LPCWSTR pszLinkDirectory, LPCWSTR pszLinkTarget)
+{
+   DWORD		dwRet = ERROR_SUCCESS;
+   // Size assumption: We have to copy 2 path with each MAX_PATH long onto the structure. So we take 3 times MAX_PATH
+   char		reparseBuffer[MAX_PATH * 3];
+   WCHAR		szDirectoryName[MAX_PATH];
+   WCHAR		szTargetName[MAX_PATH];
+   PWCHAR	szFilePart;
+   DWORD		dwLength;
+
+
+   // Get the full path referenced by the target
+   if (!GetFullPathName(pszLinkTarget, MAX_PATH, szTargetName, &szFilePart))
+      return GetLastError();
+
+   // Get the full path referenced by the directory
+   if (!GetFullPathName(pszLinkDirectory, MAX_PATH, szDirectoryName, &szFilePart))
+      return GetLastError();
+
+   // Create the link - ignore errors since it might already exist
+   BOOL bDirCreated = CreateDirectory(pszLinkDirectory, NULL);
+   if (!bDirCreated) {
+      DWORD dwErr = GetLastError();
+      if (ERROR_ALREADY_EXISTS != dwErr)
+         return dwErr;
+      else {
+         // If a Junction already exists, we have to check if it points to the 
+         // same location, and if yes then return ERROR_ALREADY_EXISTS
+         wchar_t szDestination[MAX_PATH] = { 0 };
+         AppendToPath(pszLinkDirectory, szStarDotStar);

```

Comment:
I get a compiler warning here because pszLinkDirectory is passed as const but is being modified here

Reply:
Yes .... For now. With #308 we would get a DecodeReparsePoint(), which has not the need to Append *.* to the destination, and then this line AppendTpPath() is removed. 
In my combined branch _hermann this is all gone.
Anyhow to avoid this warning in this PR I copied pszLinkDirectory temporarily over